File: ch4/clip4/boj11047/main.cpp
#include <iostream>

int main()
{
    int N, K;
    int coin[10];

    scanf("%d %d", &N, &K);

    for(int i = 0; i < N; i++)
    {
        scanf("%d", &coin[i]);
    }

    int count = 0;
    for(int i = N - 1; i >= 0; i--)
    {
        count += K / coin[i];
        K %= coin[i];
    }

    printf("%d", count);

    return 0;
}

File: ch4/clip4/boj11047/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int coin[] = new int[n];
        for (int i = 0; i < n; i++) {
            coin[i] = sc.nextInt();
        }

        int count = 0;
        for (int i = n - 1; i >= 0; i--) {
            if (k >= coin[i]) {
                count += k / coin[i];
                k %= coin[i];
            }
        }
        System.out.println(count);
    }
}
